deploy-tool/cert/cert.go

```diff
@@ -0,0 +1,84 @@
+package cert
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/x509"
+	"deploy-tool/config"
+	"deploy-tool/dns"
+	"fmt"
+	"github.com/caddyserver/certmagic"
+	"strings"
+)
+
+// IssueCertificate asks the Let's Encrypt CA to validate and sign a certificate for a site using the DNS01 strategy
+// This issue depends on multiple environment variables being set before being invoked:
+// D_CA_AGREE, expecting a value of "true" - checks to see if the user has agreed to the CA terms
+// D_CA_EMAIL, expecting an email address - this is used for the ACME account
+// D_DNS_PROVIDER, expecting a DNS provider, e.g. "cloudflare" - this is used for the DNS challenge
+// D_API_TOKEN, expecting an API token for the DNS provider - this is used to authenticate to the DNS provider
+func IssueCertificate(deployConfig config.DeployConfig, site string, privKey *ecdsa.PrivateKey) (string, error) {
+	if deployConfig.ACME.SubscriberAgreement == false {
+		return "", fmt.Errorf("let's encrypt CA terms were not agreed to")
+	}
+
+	userEmail, err := getUserEmail(deployConfig)
+	if err != nil {
+		return "", fmt.Errorf("error when getting the user email: %v", err)
+	}
+
+	dnsProvider, err := dns.GetACMEDNSProvider(deployConfig)
+
+	if err != nil {
+		return "", fmt.Errorf("error when getting the dns provider: %v", err)
+	}
+
+	certmagic.DefaultACME.DNS01Solver = &certmagic.DNS01Solver{
+		DNSProvider: dnsProvider,
+	}
+
+	certmagic.DefaultACME.Agreed = true
+	certmagic.DefaultACME.Email = userEmail
+	// TODO change this to default LE server
+	certmagic.DefaultACME.CA = certmagic.LetsEncryptStagingCA
+
+	acme := certmagic.NewDefault()
+	acme.Storage = &certmagic.FileStorage{Path: "./deploy-tool-output"}
```

Comment:
Should this specify `SecretStorage` instead of `FileStorage`?

Reply:
Actually this is probably something that I should make configurable. Just in case we need to run this locally.